pulser-pasqal/pulser_pasqal/pasqal_cloud.py

```diff
@@ -183,19 +185,27 @@ def _fetch_result(
         self, submission_id: str, job_ids: list[str] | None
     ) -> tuple[Result, ...]:
         # For now, the results are always sampled results
-        full_results = self._query_result(submission_id)
+        jobs = self._query_job_progress(submission_id)
 
         if job_ids is None:
-            job_ids = list(full_results.keys())
+            job_ids = list(jobs.keys())
 
-        results = []
+        results: list[Result] = []
         for id in job_ids:
-            assert full_results[id] is not None, "Failed to fetch the results."
-            results.append(cast(Result, full_results[id]))
+            status, result = jobs[id]
+            if status != {JobStatus.PENDING, JobStatus.RUNNING}:
```

Comment:
Didn't you mean
```suggestion
            if status in {JobStatus.PENDING, JobStatus.RUNNING}:
```
?